acvm-repo/acir/src/native_types/expression/mod.rs

```diff
@@ -273,6 +273,64 @@ impl<F: AcirField> Expression<F> {
 
         Expression { mul_terms, linear_combinations, q_c }
     }
+
+    /// Determine the width of this expression.
+    /// The width meaning the number of unique witnesses needed for this expression.
+    pub fn width(&self) -> usize {
+        let mut width = 0;
+
+        if self.mul_terms.is_empty() {
```

Comment:
I think you can simply remove this if self.mul_terms.is_empty() {..} altogether. The loop over mul terms after will do nothing if they are empty anyways

Reply:
done